lisp/ess-help.el

```diff
@@ -296,7 +308,7 @@ if necessary.  It is bound to RET and C-m in R-index pages."
             ess-local-process-name pname)
       (setq buffer-read-only nil)
       (delete-region (point-min) (point-max))
-      (ess-help-mode)
+      (ess--set-help-major-mode)
```

Comment:
Is this ess--set-help-major-mode extra step necessary? It was clear before that ess-help-mode is installed. 

If it's really needed then a better name could be `ess--install-help-mode`.

Reply:
I /really/ don't like "install" here. If you don't like set, maybe one of these instead?: 

- switch
- pick
- choose

alternatively, we could just eliminate that word and do: `ess--help-major-mode`.

Reply:
Let's just do `ess--help-mode`. But again, what's the point of having that extra step?

Reply:
OK, I'll change it. 

The point is because we have ess-help-mode and also ess-r-help-mode (and ess-stata-help-mode but it's the same as ess-help-mode at the moment). We have to decide which help mode to use and this was the simplest way I could think of when I wrote it. Do you have a different idea?

Reply:
Ok. I found the definition of that function :) I still think "install" is the best mnemonic but never-mind. That kind of dispatch we should get rid anyhow in a short run. 

>  Do you have a different idea?

Yes. We could use cl-generic and dispatch on ess-dialect, but that's orthogonal to this PR.

Reply:
When I wrote this I was worried that we'd get to a place where we'd need to set the proper help major mode before ess-dialect was set but that is apparently not the case so should be easy enough to do after this gets merged.